core/src/main/scala/compstak/kafkastreams4s/STable.scala

```diff
@@ -21,22 +21,23 @@ class STable[C[_]: Codec, K: C, V: C](val toKTable: KTable[K, V]) {
     Sync[F].delay(
       toKTable.toStream.to(
         outputTopicName,
-        producedForCodec[C, K, V]
+        producedForCodec[HK, K, HV, V]
       )
     )
 
   /**
    * Like `to`, but will render `None` as `null` in the output topic.
    */
-  def toRenderOption[F[_]: Sync, V2: C](outputTopicName: String)(implicit ev: V =:= Option[V2]): F[Unit] =
+  def toRenderOption[F[_]: Sync, V2: HV](outputTopicName: String)(implicit ev: V =:= Option[V2]): F[Unit] =
     Sync[F].delay(
       toKTable
         .toStream()
         .to(
           outputTopicName,
           Produced.`with`(
-            Codec[C].serde[K],
-            serdeForNull[C, V2].asInstanceOf[Serde[V]]
+            Codec[HK].serde[K],
+            serdeForNull[HV, V2].asInstanceOf[Serde[V]] // LIES
```

Comment:
This isn't technically a lie, because `serdeForNull[HV, V2] returns a `Serde[Option[V2]]` which given `ev` we know to be the same as `Serde[V]`. Scala 2.13 adds some nice properties to `=:=` which would allow us to remove this cast in a safe way, but unfortunately it hasn't been backported to Scala 2.12.